File: app/src/main/java/net/cynreub/subly/domain/model/PaymentMethod.kt
package net.cynreub.subly.domain.model

import java.util.UUID

data class PaymentMethod(
    val id: UUID = UUID.randomUUID(),
    val nickname: String,
    val type: PaymentType,
    val lastFourDigits: String? = null,
    val icon: Int? = null
)

File: subly_ios/Subly/Subly/Domain/Models/PaymentMethod.swift
import Foundation

struct PaymentMethod: Identifiable, Equatable, Codable {
    var id: UUID
    var nickname: String
    var type: PaymentType
    var lastFourDigits: String?

    init(
        id: UUID = UUID(),
        nickname: String,
        type: PaymentType,
        lastFourDigits: String? = nil
    ) {
        self.id = id
        self.nickname = nickname
        self.type = type
        self.lastFourDigits = lastFourDigits
    }
}
